modules/oe_newsroom_newsletter/src/Api/NewsroomMessengerBase.php

```diff
@@ -0,0 +1,360 @@
+    }
+
+    $this->privateKey = $privateKey;
+    $this->hashMethod = $hashMethod;
+    $this->normalized = $normalized;
+    $this->universe = $universe;
+    $this->app = $app;
+  }
+
+  /**
+   * Current class configuration.
+   *
+   * @codingStandardsIgnoreStart
+   * @return array{hashMethod: string, normalized: bool, universe: string, app: string}
+   *   Hash method, normalized, universe and app.
+   * @codingStandardsIgnoreEnd
+   */
+  public function getConfiguration() {
+    return [
+      'hashMethod' => $this->hashMethod,
+      'normalized' => $this->normalized,
+      'universe' => $this->universe,
+      'app' => $this->app,
+    ];
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function subscriptionServiceConfigured(bool $throw_error = TRUE): bool {
+    if (!$throw_error) {
+      return $this->newsroomApiUsable();
+    }
+
+    if (!$this->newsroomApiUsable()) {
+      throw new InvalidApiConfiguration($this->t('The subscription service is not configured at the moment. Please try again later.')->render());
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * Generates a key from the e-mail and from the private key.
+   *
+   * @param string $email
+   *   Subscriber e-mail.
+   *
+   * @return string
+   *   Generated communication key.
+   */
+  protected function generateKey(string $email): string {
+    if ($this->normalized) {
+      return hash($this->hashMethod, mb_strtolower($email) . $this->privateKey);
+    }
+
+    return hash($this->hashMethod, $email . $this->privateKey);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function subscribe(string $email, array $svIds = [], array $relatedSvIds = [], string $language = NULL, array $topicExtId = []) : ?array {
+    $this->subscriptionServiceConfigured();
+
+    // Prepare the post.
+    $input = [
+      'key' => $this->generateKey($email),
+      'subscription' => [
+        'universeAcronym' => $this->universe,
+        'topicExtWebsite' => $this->app,
+        'sv_id' => implode(',', $svIds),
+        'email' => $this->normalized ? mb_strtolower($email) : $email,
+        'language' => $language,
+      ],
+    ];
+
+    if (!empty($relatedSvIds)) {
+      $input['subscription']['relatedSv_Id'] = implode(',', $relatedSvIds);
+    }
+    if (!empty($topicExtId)) {
+      $input['subscription']['topicExtId'] = implode(',', $topicExtId);
+    }
+
+    try {
+      // Send the request.
+      $request = $this->httpClient->request('POST', $this->subscriptionSubscribeUrl, ['json' => $input]);
+      if ($request->getStatusCode() === 200) {
+        $data = Json::decode($request->getBody()->getContents());
+
+        if (empty($data)) {
+          throw new BadResponseException($this->t('Empty response returned by Newsroom newsletter API.')->render(), NULL);
+        }
+
+        $response = NULL;
+        // This is necessary to split separately newsletters distribion lists.
+        $sv_ids_separated = explode(',', implode(',', $svIds));
```

Comment:
Yes. An sv id is only a number. If there is a comma between them, it means there is more than one sv id in that string. When you want to group multiple distribution lists under a single label, this will split it up.